core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -116,16 +116,16 @@ class BpmBillingProjectLifecycle(
 
   override def preDeletionSteps(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext): Future[Unit] =
     for {
-      _ <- billingRepository.getCreationStatus(projectName).map {
+      _ <- billingRepository.getCreationStatus(projectName).flatMap {
```

Comment:
These two are equivalent and a matter of personal taste. I think Blake suggested not using `flatMap` here as
- you weren't calling another function that returned a `Future` and suggested instead could use `map` and return `()`.
- `flatMap`-ing in a `for`-comp is a bit odd as each subsequent statement (`res <- expr`) is actually syntactic sugar for `flatMap`.

Reply:
As an aside, I prefer the use of `flatMap` because it captures the idea that the following expression has side effects - namely, you're throwing an exception. In Scala, throwing in the in the context of `Future` with `map` or `flatMap`, however, is perfectly legal.
I could go on but I'm not sure how productive it'll be for fear it'll just lead to more confusion!

Reply:
Hmmm, I unfortunately don't see the difference between lines 119 and 150. In both cases, the result of the function call is a Future. The return type of  `getCreationStatus` is `Future[CreationStatus]`, whereas the return type of `getBillingProfileId` is ` Future[Option[String]]`. I was hoping for a general rule like "use flatMap" whenever you are dealing with Futures". Is the issue here that the call to `getBillingProfileId` is the last call in the for comprehension??

Reply:
I meant to post my long reply to this. See below - hope it helps :)